app/controllers/shopify_app/sessions_controller.rb

```diff
@@ -35,14 +35,19 @@ def destroy
 
     def authenticate
       if sanitized_shop_name.present?
-        fullpage_redirect_to "#{main_app.root_path}auth/shopify?shop=#{sanitized_shop_name}"
+        session['shopify.omniauth_params'] = { shop: sanitized_shop_name }
```

Comment:
from the other PR there is some concern that this won't work in Safari. We need to test and confirm.